addons/website_sale_wallet/models/partner.py

```diff
@@ -0,0 +1,80 @@
+    order_line_gift_card_ids = fields.One2many("sale.order.line", "wallet_id")
+    currency_id = fields.Many2one('res.currency',
+                                  string='Currency',
+                                  readonly=True,
+                                  default=lambda self: self.env.company.currency_id.id,
+                                  required=True)
+
+    # Computed fields
+    balance = fields.Monetary(compute="_compute_balance", currency_field='currency_id')
+    actual_balance = fields.Monetary(compute="_compute_balance", currency_field='currency_id')
+    initial_balance = fields.Monetary(compute="_compute_initial_balance", currency_field='currency_id')
+    current_website = fields.Many2one('website', compute='_compute_current_website')
+
+    # Computed func
+    def _compute_current_website(self):
+        self.current_website = self.env['website'].get_current_website()
+
+    @api.depends("gift_card_ids", "current_website")
+    def _compute_initial_balance(self):
+        for record in self:
+            record.initial_balance = functools.reduce(
+                lambda value, gift_card: value + gift_card.initial_balance,
+                record.gift_card_ids.filtered(
+                    lambda gift_card: gift_card.website_id == self.current_website
+                ), 0
+            )
+
+    @api.depends('order_line_gift_card_ids', "current_website")
+    def _compute_balance(self):
+        for record in self:
+            order_line_filtered_current_website = record.order_line_gift_card_ids.filtered(
+                lambda order_line: order_line.website_id == self.current_website)
+            record.actual_balance = record.initial_balance + functools.reduce(
+                lambda o1, o2: (o1 + o2.price_unit * o2.product_uom_qty),
+                order_line_filtered_current_website, 0
+            )
+            record.balance = record.initial_balance + functools.reduce(
+                lambda o1, o2: (o1 + o2.price_unit * o2.product_uom_qty),
+                order_line_filtered_current_website.filtered(lambda order_line: order_line.state == 'sale'), 0
+            )
+            self._release_not_confirmed_orders(record)
+
+    def _release_not_confirmed_orders(self, record):
+        if record.actual_balance < 0:
+            actual_balance = {
+                'actual_balance': record.actual_balance
+            }
+            order_line_filtered_state_not_sale = record.order_line_gift_card_ids.filtered(
+                lambda o_l: o_l.website_id == self.current_website
+                and o_l.state != "sale"
+                and self.assign_and_return(o_l.price_unit * o_l.product_uom_qty, actual_balance) >= 0
+            )
+            record.actual_balance = 0
+            order_line_filtered_state_not_sale.product_uom_qty = 0
+
+    @staticmethod
+    def assign_and_return(price, actual_balance):
+        actual_balance['actual_balance'] -= price
+        return actual_balance['actual_balance']
```

Comment:
strang way to write it

Reply:
New to python 😅
I want an Integer object (not primitive type), to use it in lambda expression.